pylint/checkers/utils.py

```diff
@@ -1457,21 +1481,27 @@ def is_registered_in_singledispatch_function(node: nodes.FunctionDef) -> bool:
 
     decorators = node.decorators.nodes if node.decorators else []
     for decorator in decorators:
-        # func.register are function calls
-        if not isinstance(decorator, nodes.Call):
-            continue
+        if func_def := find_infered_fn_from_register(decorator):
+            return decorated_with(func_def, singledispatch_qnames)
 
-        func = decorator.func
-        if not isinstance(func, nodes.Attribute) or func.attrname != "register":
-            continue
+    return False
 
-        try:
-            func_def = next(func.expr.infer())
-        except astroid.InferenceError:
-            continue
 
-        if isinstance(func_def, nodes.FunctionDef):
-            return decorated_with(func_def, singledispatch_qnames)
+def is_registered_in_singledispatchmethod_function(node: nodes.FunctionDef) -> bool:
```

Comment:
Perhaps put the refactoring in another PR. That would help reviewing this and keep the size of the PR smaller.

Reply:
Just to make sure I understand, by refactoring you mean the common code i moved to `find_infered_fn_from_register`? or not touching `pylint/checkers/utils.py` at all and add the logic for resolving `singledispatch` and `singledispatchmethod` in the stdlib checker module?

Reply:
Yeah, the first thing. Let's focus this PR on everything that is strictly necessary for the new messages. Anything else can be moved to separate PRs as it can be easily approved and merged.